request/src/main/java/com/vimeo/networking2/internal/VimeoApiClientImpl.kt

```diff
@@ -0,0 +1,835 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.vimeo.networking2.internal
+
+import com.vimeo.networking2.*
+import com.vimeo.networking2.common.Followable
+import com.vimeo.networking2.config.Configuration
+import com.vimeo.networking2.enums.CommentPrivacyType
+import com.vimeo.networking2.enums.ConnectedAppType
+import com.vimeo.networking2.enums.EmbedPrivacyType
+import com.vimeo.networking2.enums.ErrorCodeType
+import com.vimeo.networking2.enums.ViewPrivacyType
+import com.vimeo.networking2.params.BatchPublishToSocialMedia
+import com.vimeo.networking2.params.ModifyVideoInAlbumsSpecs
+import com.vimeo.networking2.params.ModifyVideosInAlbumSpecs
+import com.vimeo.networking2.params.SearchDateType
+import com.vimeo.networking2.params.SearchDurationType
+import com.vimeo.networking2.params.SearchFacetType
+import com.vimeo.networking2.params.SearchFilterType
+import com.vimeo.networking2.params.SearchSortDirectionType
+import com.vimeo.networking2.params.SearchSortType
+import okhttp3.CacheControl
+
+/**
+ * The internal implementation of the [VimeoApiClient]. Once configured, this client cannot be re-configured.
+ *
+ * @param vimeoService The service used to make requests to the Vimeo API.
+ * @param authenticator The authenticator used to obtain tokens which can be used to make requests.
+ * @param configuration The configuration used by this client instance.
+ * @param basicAuthHeader The basic auth header using the client ID and secret, used if the account store does not
+ * provide an authenticated account.
+ * @param localVimeoCallAdapter The adapter used to notify consumers of local errors.
+ */
+@Suppress("LargeClass")
+internal class VimeoApiClientImpl(
+    private val vimeoService: VimeoService,
+    private val authenticator: Authenticator,
+    private val configuration: Configuration,
+    private val basicAuthHeader: String,
+    private val localVimeoCallAdapter: LocalVimeoCallAdapter
+) : VimeoApiClient {
+
+    private val authHeader: String
+        get() {
+            val vimeoAccount = authenticator.currentAccount
+
+            return vimeoAccount?.accessToken?.let { "Bearer $it" } ?: basicAuthHeader
+        }
+
+    override fun createAlbum(
```

Comment:
Do we have plans to group and move related APIs (e.g. Album) out of VimeoApiClient into some encapsulating class? This could help consumers to navigate the API features more easily. We could group the features in a similar way to the organization that is presented in the API reference: https://developer.vimeo.com/api/reference

Reply:
I was hoping to, but at this point the lines are not clear enough to justify the increased complexity. There are a couple features that have clear boundaries like albums and connected apps, but there most parts of the API are not fully developed. For example, channels utilize generic `fetchVideo`, `putContent`, and `deleteContent` functions. In order to create a "channels client," we would have to create separate `addToChannel`, `removeFromChannel`, `fetchVideoInChannel` functions. It's not feasible for us to do right now given time constraints. 

Given our limited resources, I'm not sure that it's completely responsible for us to do this in the near term, as it will increase the overall complexity (while improving discoverability) and will increase the maintenance costs associated with the library due to an increased number of classes and (presumably) new rules about adding new endpoints.